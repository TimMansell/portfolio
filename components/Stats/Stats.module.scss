@import "styles/scssvars";
@import "styles/utilities/pxtorem";
@import "styles/utilities/flex-gap";

.lighterNote {
  display: grid;
  color: var(--secondary-text-colour);

  @media (min-width: $bp-sm) {
    grid-gap: var(--vr-lg);
    grid-template-columns: repeat(2, 1fr);
  }

  @media (min-width: $bp-lg) {
    grid-gap: var(--vr-md);
    grid-template-columns: repeat(4, 1fr);
  }
}

.container {
  composes: divider from "styles/helpers/divider.scss";
}

.item {
  @include flex-gap(--vr);

  display: flex;
  flex-direction: row;
  align-items: flex-start;
  justify-self: center;
}

.title {
  font-size: pxtorem(42);
  line-height: 0.8;
  margin-bottom: var(--vr-xxs);
}

.content {
  width: 100%;

  @media (min-width: $bp-sm) {
    margin-left: var(--vr-xs);
  }
}

.icon {
  display: flex;
  justify-content: center;
}

.description {
  font-size: pxtorem(18);
  margin: 0;
}
