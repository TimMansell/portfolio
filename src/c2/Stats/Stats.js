import React from 'react';

import InViewport from '../InViewport';
import Counter from '../Counter';

import { library } from '@fortawesome/fontawesome-svg-core';
import { faCode, faCodeBranch, faMugHot, faBicycle } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

library.add(faCode, faCodeBranch, faMugHot, faBicycle);

export const Stats = () => {
	return <>
		<div className="lighter-note">
			<div className="lighter-note__item divider">
				<FontAwesomeIcon icon={faCode} className="icon" />
				<InViewport>
					<p className="lighter-note__amount">
						<Counter begin={3000} end={50000} time={2000} />
					</p>
				</InViewport>
				<p className="lighter-note__text">lines of code</p>
			</div>
			<div className="lighter-note__item divider">
				<FontAwesomeIcon icon={faCodeBranch} className="icon" />
				<InViewport>
					<p className="lighter-note__amount">
						<Counter begin={300} end={1200} time={2000} />
					</p>
				</InViewport>
				<p className="lighter-note__text">git commits</p>
			</div>
			<div className="lighter-note__item divider">
				<FontAwesomeIcon icon={faMugHot} className="icon" />
				<InViewport>
					<p className="lighter-note__amount">
						<Counter begin={200} end={760} time={2000} />
					</p>
				</InViewport>
				<p className="lighter-note__text">coffees consumed</p>
			</div>
			<div className="lighter-note__item divider">
				<FontAwesomeIcon icon={faBicycle} className="icon" />
				<InViewport>
					<p className="lighter-note__amount">
						<Counter begin={650} end={3500} time={2000} />
					</p>
				</InViewport>
				<p className="lighter-note__text">kilometers cycled</p>
			</div>
		</div>
	</>;
}

export default Stats;