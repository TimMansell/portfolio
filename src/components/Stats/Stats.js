import React, { useRef } from 'react';

import Header from 'components/Header';
import Counter from 'components/Counter';

import useInViewport from 'hooks/useInViewport';

import stats from './json/stats.json';

import styles from './Stats.module.scss';

import {
  IconCode,
  IconCodeBranch,
  IconCodeLaptop,
  IconMugHot,
} from 'components/Icon';

export const Stats = () => {
  const element = useRef(null);
  const isInViewport = useInViewport(element, -300);

  return (
    <>
      <Header title="Stats" text="This website was built using" tertiary />
      <div className={styles.lighterNote} ref={element}>
        <div className={styles.item}>
          <IconCode padded />
          <p className={styles.amount}>
            {isInViewport && (
              <Counter
                begin={Math.round(stats.loc * 0.9)}
                end={stats.loc}
                time={2000}
              />
            )}
          </p>
          <p className={styles.text}>lines of code</p>
        </div>
        <div className={styles.item}>
          <IconCodeBranch padded />
          <p className={styles.amount}>
            {isInViewport && (
              <Counter
                begin={Math.round(stats.commits * 0.5)}
                end={stats.commits}
                time={2000}
              />
            )}
          </p>
          <p className={styles.text}>git commits</p>
        </div>
        <div className={styles.item}>
          <IconCodeLaptop padded />
          <p className={styles.amount}>
            {isInViewport && (
              <Counter begin={10} end={stats.pullRequests} time={2000} />
            )}
          </p>
          <p className={styles.text}>pull requests</p>
        </div>
        <div className={styles.item}>
          <IconMugHot padded />
          <p className={styles.amount}>
            {isInViewport && (
              <Counter begin={0} end={stats.coffees} time={2000} />
            )}
          </p>
          <p className={styles.text}>coffees</p>
        </div>
      </div>
    </>
  );
};

export default Stats;
