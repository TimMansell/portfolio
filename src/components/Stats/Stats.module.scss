@import 'variables';
@import 'utilities/pxtorem';

.lighterNote {
  display: grid;
  grid-gap: 16px;

  @media (min-width: $bp-sm) {
    grid-template-columns: repeat(4, 1fr);
  }
}

.item {
  composes: divider from '~scss/helpers/divider.scss';
  color: $secondary-text-colour;
}

.amount {
  font-size: pxtorem(40);
  text-shadow: -1px -1px darken($primary-colour, 10%);
}

.text {
  text-shadow: -1px -1px darken($primary-colour, 10%);
  text-align: center;
}
