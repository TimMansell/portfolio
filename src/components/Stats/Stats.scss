@import 'variables';
@import 'utilities/breakpoints';

.lighter-note {
	display: grid;
	grid-gap: 16px;

	@include media-breakpoint-up(sm){
		grid-template-columns: repeat(4, 1fr);
	}

	&__item {
		display: flex;
		flex-direction: column;
		align-items: center;
		color: $secondary-text-colour;
	}

	&__amount {
		font-size: 40px;
  		text-shadow: -1px -1px darken($primary-colour, 10%);
	}

	&__text {
		text-shadow: -1px -1px darken($primary-colour, 10%);
		text-align: center;
	}
}