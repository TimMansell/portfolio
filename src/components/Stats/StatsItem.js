import React, { useRef } from 'react';
import PropTypes from 'prop-types';

import Icon from 'components/Icon';
import Counter from 'components/Counter';

import useIntersectionObserver from 'hooks/useIntersectionObserver';

import styles from './Stats.module.scss';

export const offsetStats = (value, offset) => Math.round(value * offset);

export const StatsItem = ({ description, offset, value, icon }) => {
  const element = useRef(null);
  const isInViewport = useIntersectionObserver(element, { offset: -250 });
  const statsBeginValue = offsetStats(value, offset);

  return (
    <div className={styles.container} data-testid="stats-item">
      <div className={styles.item} ref={element}>
        <div className={styles.icon}>
          <Icon name={['fa', icon]} size="sm" />
        </div>
        <div className={styles.content}>
          <h3 className={styles.title}>
            {isInViewport && (
              <Counter begin={statsBeginValue} end={value} time={2000} />
            )}
            {!isInViewport && (
              <span data-e2e="stats-starting-count"> {statsBeginValue}</span>
            )}
          </h3>
          <p className={styles.description}>{description}</p>
        </div>
      </div>
    </div>
  );
};

StatsItem.propTypes = {
  description: PropTypes.string.isRequired,
  offset: PropTypes.number.isRequired,
  value: PropTypes.number.isRequired,
  icon: PropTypes.string.isRequired,
};

export default StatsItem;
