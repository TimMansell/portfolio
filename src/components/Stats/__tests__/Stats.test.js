import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Stats from '../Stats';

describe('Stats', () => {
  it('should render my component', () => {
    const wrapper = shallow(<Stats />);
  });

  it('should match snapshot', () => {
    const snapshot = renderer.create(<Stats/>).toJSON();

    expect(snapshot).toMatchSnapshot();
  });
});
