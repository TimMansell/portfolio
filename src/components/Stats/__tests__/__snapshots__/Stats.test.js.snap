// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats should match snapshot 1`] = `
<div
  className="lighterNote"
>
  <div
    className="item"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-code fa-w-20 icon padded"
      data-icon="code"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 640 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M278.9 511.5l-61-17.7c-6.4-1.8-10-8.5-8.2-14.9L346.2 8.7c1.8-6.4 8.5-10 14.9-8.2l61 17.7c6.4 1.8 10 8.5 8.2 14.9L293.8 503.3c-1.9 6.4-8.5 10.1-14.9 8.2zm-114-112.2l43.5-46.4c4.6-4.9 4.3-12.7-.8-17.2L117 256l90.6-79.7c5.1-4.5 5.5-12.3.8-17.2l-43.5-46.4c-4.5-4.8-12.1-5.1-17-.5L3.8 247.2c-5.1 4.7-5.1 12.8 0 17.5l144.1 135.1c4.9 4.6 12.5 4.4 17-.5zm327.2.6l144.1-135.1c5.1-4.7 5.1-12.8 0-17.5L492.1 112.1c-4.8-4.5-12.4-4.3-17 .5L431.6 159c-4.6 4.9-4.3 12.7.8 17.2L523 256l-90.6 79.7c-5.1 4.5-5.5 12.3-.8 17.2l43.5 46.4c4.5 4.9 12.1 5.1 17 .6z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <div>
      <span>
         
      </span>
    </div>
    <p
      className="text"
    >
      lines of code
    </p>
  </div>
  <div
    className="item"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-code-branch fa-w-12 icon padded"
      data-icon="code-branch"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 384 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M384 144c0-44.2-35.8-80-80-80s-80 35.8-80 80c0 36.4 24.3 67.1 57.5 76.8-.6 16.1-4.2 28.5-11 36.9-15.4 19.2-49.3 22.4-85.2 25.7-28.2 2.6-57.4 5.4-81.3 16.9v-144c32.5-10.2 56-40.5 56-76.3 0-44.2-35.8-80-80-80S0 35.8 0 80c0 35.8 23.5 66.1 56 76.3v199.3C23.5 365.9 0 396.2 0 432c0 44.2 35.8 80 80 80s80-35.8 80-80c0-34-21.2-63.1-51.2-74.6 3.1-5.2 7.8-9.8 14.9-13.4 16.2-8.2 40.4-10.4 66.1-12.8 42.2-3.9 90-8.4 118.2-43.4 14-17.4 21.1-39.8 21.6-67.9 31.6-10.8 54.4-40.7 54.4-75.9zM80 64c8.8 0 16 7.2 16 16s-7.2 16-16 16-16-7.2-16-16 7.2-16 16-16zm0 384c-8.8 0-16-7.2-16-16s7.2-16 16-16 16 7.2 16 16-7.2 16-16 16zm224-320c8.8 0 16 7.2 16 16s-7.2 16-16 16-16-7.2-16-16 7.2-16 16-16z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <div>
      <span>
         
      </span>
    </div>
    <p
      className="text"
    >
      git commits
    </p>
  </div>
  <div
    className="item"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-mug-hot fa-w-16 icon padded"
      data-icon="mug-hot"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M127.1 146.5c1.3 7.7 8 13.5 16 13.5h16.5c9.8 0 17.6-8.5 16.3-18-3.8-28.2-16.4-54.2-36.6-74.7-14.4-14.7-23.6-33.3-26.4-53.5C111.8 5.9 105 0 96.8 0H80.4C70.6 0 63 8.5 64.1 18c3.9 31.9 18 61.3 40.6 84.4 12 12.2 19.7 27.5 22.4 44.1zm112 0c1.3 7.7 8 13.5 16 13.5h16.5c9.8 0 17.6-8.5 16.3-18-3.8-28.2-16.4-54.2-36.6-74.7-14.4-14.7-23.6-33.3-26.4-53.5C223.8 5.9 217 0 208.8 0h-16.4c-9.8 0-17.5 8.5-16.3 18 3.9 31.9 18 61.3 40.6 84.4 12 12.2 19.7 27.5 22.4 44.1zM400 192H32c-17.7 0-32 14.3-32 32v192c0 53 43 96 96 96h192c53 0 96-43 96-96h16c61.8 0 112-50.2 112-112s-50.2-112-112-112zm0 160h-16v-96h16c26.5 0 48 21.5 48 48s-21.5 48-48 48z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <div>
      <span>
         
      </span>
    </div>
    <p
      className="text"
    >
      coffees consumed
    </p>
  </div>
  <div
    className="item"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-bicycle fa-w-20 icon padded"
      data-icon="bicycle"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 640 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M512.509 192.001c-16.373-.064-32.03 2.955-46.436 8.495l-77.68-125.153A24 24 0 0 0 368.001 64h-64c-8.837 0-16 7.163-16 16v16c0 8.837 7.163 16 16 16h50.649l14.896 24H256.002v-16c0-8.837-7.163-16-16-16h-87.459c-13.441 0-24.777 10.999-24.536 24.437.232 13.044 10.876 23.563 23.995 23.563h48.726l-29.417 47.52c-13.433-4.83-27.904-7.483-42.992-7.52C58.094 191.83.412 249.012.002 319.236-.413 390.279 57.055 448 128.002 448c59.642 0 109.758-40.793 123.967-96h52.033a24 24 0 0 0 20.406-11.367L410.37 201.77l14.938 24.067c-25.455 23.448-41.385 57.081-41.307 94.437.145 68.833 57.899 127.051 126.729 127.719 70.606.685 128.181-55.803 129.255-125.996 1.086-70.941-56.526-129.72-127.476-129.996zM186.75 265.772c9.727 10.529 16.673 23.661 19.642 38.228h-43.306l23.664-38.228zM128.002 400c-44.112 0-80-35.888-80-80s35.888-80 80-80c5.869 0 11.586.653 17.099 1.859l-45.505 73.509C89.715 331.327 101.213 352 120.002 352h81.3c-12.37 28.225-40.562 48-73.3 48zm162.63-96h-35.624c-3.96-31.756-19.556-59.894-42.383-80.026L237.371 184h127.547l-74.286 120zm217.057 95.886c-41.036-2.165-74.049-35.692-75.627-76.755-.812-21.121 6.633-40.518 19.335-55.263l44.433 71.586c4.66 7.508 14.524 9.816 22.032 5.156l13.594-8.437c7.508-4.66 9.817-14.524 5.156-22.032l-44.468-71.643a79.901 79.901 0 0 1 19.858-2.497c44.112 0 80 35.888 80 80-.001 45.54-38.252 82.316-84.313 79.885z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <div>
      <span>
         
      </span>
    </div>
    <p
      className="text"
    >
      kilometers cycled
    </p>
  </div>
</div>
`;
